EIPS/eip-4910.md

```diff
@@ -0,0 +1,1109 @@
+```
+/// @dev Function to payout a royalty payment
+/// @param tokenId is the identifier of the NFT token
+/// @param RAsubaccount is the address of the Royalty Sub Account from which the payout should happen
+/// @param receiver is the address to receive the payout
+/// @param amount is the amount to be paid out
+
+function royaltyPayOut (uint256 tokenId, address RAsubaccount, address payable payoutAccount, uint256 amount) public virtual nonReentrant returns (bool)
+```
+The Boolean return value is `true` for a successful function execution, and `false` for an unsuccessful function execution.
+
+Note, that the function has `reentrancy` protection through `nonReentrant` from the [Open Zeppelin library](https://github.com/OpenZeppelin/openzeppelin-contracts/blob/master/contracts/security/ReentrancyGuard.sol) since funds are being paid out.
+
+#### **[R59]**
+
+*The input parameters of the `royaltyPayOut` function MUST satisfy the following requirements:*
+* *`_msgSender() == RAsubaccount`.*
+* *`tokenId` must exist and MUST NOT be burned.*
+* *`tokenId` must be associated with a Royalty Account.*
+* *`RAsubaccount` must be a valid `accountId` in a Royalty Sub Account of the Royalty Account of the `tokenId'.*
+* *`isIndividual == true` for the Royalty Sub Account, `RAsubaccount`.*
+* *`amount <= royaltybalance` of the Royalty Sub Account, `RAsubaccount.*`
+
+#### **[R60]**
+
+*The internal `_royaltyPayOut` function interface MUST adhere to the definition below*:
+```
+function _royaltyPayOut (uint256 tokenId, address RAsubaccount, address payable payoutAccount, uint256 amount) public virtual returns (bool)
+```
+#### **[R61]**
+
+*The internal `_royaltyPayOut` function MUST perform the following actions:
+* *send the payment to the `payoutaccount`.*
+* *update the `royaltybalance` of the `RAsubaccount` of the Royalty Account upon successful transfer.*
+
+#### Sending a Payout Payment
+
+#### **[R62]**
+
+*The following steps MUST be taken to send out a royalty payment to its recipient:*
+* *find the Royalty Sub Account.*
+* *extract `tokentype` from the Royalty Sub Account.*
+* *based on the token type send to the `payoutAccount` either*
+    * *'ETH' / relevant protocol token or*
+    * *another token based on token type* 
+* *and only if the payout transaction is successful, deduct `amount` from `royaltybalance` of the Royalty Sub Account,`RAsubaccount`, and then return `true` as the function return parameter, otherwise return `false`.* 
+
+## Backwards Compatibility
+This EIP is backwards compatible to the ERC721 standard introducing new interfaces and functionality but retaining the core interfaces and functionality of the ERC-721 standard.
+
+## Test Cases
+A full test suite is part of the reference implementation.
+
+## Reference Implementation
+The Treetrunk reference implementation of the standard can be found [here](https://github.com/treetrunkio/treetrunk-nft-reference-implementation).
```

Comment:
The reference implementation should be included in the `assets` folder. See EIP-1 for details.

Reply:
done

Reply:
could not add the entire repo as a subfolder. Opted to add an md file with a link to the reference implementation. @lightclient

Reply:
You should not be including external links anywhere in the EIP.  EIPs should stand on their own without external dependencies.  The Reference Implementation is optional, so you can just leave it out if it doesn't make sense to have a reference implementation.

Generally speaking, if the EIP is too complex to write a simple reference implementation (it doesn't have to be production worthy, it should focus on readability) that suggests the standard may be too complex.  This isn't always true, but it is a reasonable hint.

Reply:
@MicahZoltu I got an error on commit when adding a link to a folder under `../assets/eip-4910`. I will simply add the reference implementation folder and link to the eip-4910 folder under assets in the md file. The reference implementation is very instructive for other implementers and contains full test case coverage with extensive tooling and security tests that will help implementers.

Hence, it is not a question of size or complexity of the reference implementation but rather how to refer to it.